WordPress/Classes/Services/ReaderTopicService+FollowedSites.swift

```
@@ -0,0 +1,66 @@
+
+extension ReaderTopicService {
+
+    enum FollowedSitesError: Error {
+        case unknown
+    }
+
+    @objc func fetchAllFollowedSites(success: @escaping () -> Void, failure: @escaping (Error?) -> Void) {
+        let service = ReaderTopicServiceRemote(wordPressComRestApi: apiRequest())
+        let pageSize: UInt = 100
+
+        service.fetchFollowedSites(forPage: 1, number: pageSize) { totalSites, sites in
+            guard let totalSites, let sites else {
+                failure(nil)
+                return
+            }
+            let totalPages = Int(ceil(Double(totalSites.intValue) / Double(pageSize)))
+            WPAnalytics.subscriptionCount = totalSites.intValue
+
+            guard totalPages > 1 else {
+                self.mergeFollowedSites(sites, withSuccess: success)
+                return
+            }
+
+            Task {
+                await withTaskGroup(of: Result<[RemoteReaderSiteInfo], Error>.self) { taskGroup in
+                    for page in 2...totalPages {
+                        taskGroup.addTask {
+                            return await self.fetchFollowedSites(service: service, page: UInt(page), number: pageSize)
+                        }
+                    }
+                    var allSites = sites
+                    for await result in taskGroup {
+                        if case .success(let sites) = result {
+                            allSites.append(contentsOf: sites)
+                        }
```

Hmm, I'm wondering what we should do if errors occur within `2...totalPages`. Currently, the errors are ignored, and this may lead to missing sites from the followed blogs list despite the successful result. But I'm not sure myself how we should address this without making it overly complicated... 😅 

Some ideas: we could store the failed page numbers to retry. But in case of network errors, it could very well throw the user into a loop of multi-requests, especially considering that `fetchAllFollowedSites` are getting called multiple times after the app launches.

Anyways, it's still better than just fetching the first page! 😄

---

On Android, we're clearing the local sites list and replacing it with new sites fetched from the remote server. To keep the logic straightforward in this initial iteration, we've assumed that a network chain is successful only if all paged network calls are completed successfully. At the conclusion of the last call, we consider it a success, store the results locally, and emit an event to update the UI. If an issue arises in any of the calls, the entire chain is marked as a failure, and the local sites are NOT updated. 

I suggest keeping the logic as simple as possible. Imagine you have 100+ sites in the app. The Sites pill indicates 100+ sites, and at some point, you perform a pull to refresh, or the app updates the sites upon opening the Reader. If an error occurs during a middle call, you will see fewer sites than before without a clear explanation.

---

I'll update it to fail if one of the pages doesn't succeed. It'll just use the first error we encounter.

Which error does Android use if multiple of the calls fail? I could see the logic getting pretty complicated if we tried to take each into account.

---

Similarly on Android, we log the first error encountered, even though we don't actually propagate the error to the UI.
```
            public void onErrorResponse(VolleyError volleyError) {
                AppLog.e(AppLog.T.READER, volleyError);
                serverBlogs.clear();
                taskCompleted(UpdateTask.FOLLOWED_BLOGS);
            }
```